src/mdisasm.c

```diff
@@ -1,6 +1,195 @@
 #include <stdio.h>
 #include <stdlib.h>
 
+#define OPCODE_P(opcode) (opcode >> 12)
+#define OPCODE_X(opcode) ((opcode & 0xF00) >> 8)
+#define OPCODE_Y(opcode) ((opcode & 0xF0) >> 4)
+#define OPCODE_N(opcode) (opcode & 0xF)
+#define OPCODE_NNN(opcode) (opcode & 0xFFF)
+#define OPCODE_KK(opcode) (opcode & 0xFF)
+
+static void
+to_instruction_0(unsigned short opcode, char* out)
+{
+    if (opcode == 0x00E0) {
+        snprintf(out, 20, "EXIT");
+    } else if (opcode == 0x00EE) {
+        snprintf(out, 20, "RET");
+    } else {
+        snprintf(out, 20, "SYS %x", OPCODE_NNN(opcode));
+    }
+}
+
+static void
+to_instruction_1(unsigned short opcode, char* out)
+{
+    snprintf(out, 20, "JP %x", OPCODE_NNN(opcode));
+}
+
+static void
+to_instruction_2(unsigned short opcode, char* out)
+{
+    snprintf(out, 20, "CALL %x", OPCODE_NNN(opcode));
+}
+
+static void
+to_instruction_3(unsigned short opcode, char* out)
+{
+    snprintf(out, 20, "SE V[%x], %x", OPCODE_X(opcode), OPCODE_KK(opcode));
+}
+
+static void 
+to_instruction_4(unsigned short opcode, char* out)
+{
+    snprintf(out, 20, "SNE V[%x], %x", OPCODE_X(opcode), OPCODE_KK(opcode));
+}
+
+static void
+to_instruction_5(unsigned short opcode, char* out)
+{
+    snprintf(out, 20, "SE V[%x], V[%x]", OPCODE_X(opcode), OPCODE_Y(opcode));
+}
+
+static void
+to_instruction_6(unsigned short opcode, char* out)
+{
+    snprintf(out, 20, "LD V[%x], %x", OPCODE_X(opcode), OPCODE_KK(opcode));
+}
+
+static void
+to_instruction_7(unsigned short opcode, char* out)
+{
+    snprintf(out, 20, "AND V[%x], %x", OPCODE_X(opcode), OPCODE_KK(opcode));
```

Comment:
It should be ADD instead of AND

http://devernay.free.fr/hacks/chip8/C8TECH10.HTM#7xkk